README.md

```diff
@@ -249,7 +249,62 @@ module.exports = {
     let checker = VersionChecker.forProject(this.project);
 
     for (let { name, root } = checker.allAddons()) {
-      // access to the add-on, in this case root + name
+      // access to the addon, in this case root + name
+    }
+  }
+};
+```
+### check
+
+A utility to verify that addons are installed at appropriate versions.
+Although `npm` and `yarn` typically manage dependencies for us, they do not
+provide a mechanism for an addon to specific a hard constraint. When `npm` or
+`yarn `see and invalid dependency, they resolve this by duplicating said
+dependency. In-general, the node ecosystem works fine with duplication;
+unfortunately, this strategy doesn't always work.
+
+Every addon in the ember ecosystem implicitly depends `ember-source` and most
+likely a specific version range. If that dependency is specified as a
+`package.json` dependency, a mismatch between application and addon would
+result in duplicating `ember-source`. This is both unhelpful and unwanted.
```

Comment:
```suggestion
likely a specific version range. If that dependency is specified as a
`package.json` dependency, a mismatch between application and addon would
result in duplicating `ember-source`.  Instead of failing the build, we would build an application with an unknown version of `ember-source`, subverting the point of specifying dependency version ranges in the first place!  The `check` API provides a mechanism to avoid this and fail fast in the build step, instead of building an invalid application with harder to debug runtime errors.
```